staging/src/k8s.io/apiserver/pkg/registry/generic/registry/decorated_watcher.go

```diff
@@ -61,20 +65,24 @@ func (d *decoratedWatcher) run(ctx context.Context) {
 			}
 			select {
 			case d.resultCh <- send:
-				if send.Type == watch.Error {
-					d.cancel()
-				}
```

Comment:
This assumed the underlying watcher closed when it sent an error event, which is not necessarily true. The decoratedWatcher pipe should be dumb and merely propagate events, decorating non-error events, until the underlying watcher closes its result channel or the context is cancelled

Reply:
When is this assumption not true today?  can you give an example?  Maybe this is just cleanup overall?

Reply:
edited my comment to clarify... all known watcher implementations do close on error